common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/service/DefaultTransportService.java

```diff
@@ -824,6 +827,21 @@ public void log(TransportProtos.SessionInfoProto sessionInfo, String msg) {
         process(sessionInfo, postTelemetryMsg, TransportServiceCallback.EMPTY);
     }
 
+    private boolean checkMsgSize(TransportProtos.SessionInfoProto sessionInfo, TransportServiceCallback<?> callback, CharSequence... payload) {
```

Comment:
The algorithm complexity should be O(1). Any higher complexity will not help prevent DDoS attacks (CPU overload).
You may sacrifice the precise calculation and use an estimate. e.g. charSequence length is enough. the real size may vary from 1 to 2 bytes per char. but it might be well enough protection.

But? please, read another comments before you made any changes.